awx/ui/client/src/users/users.form.js

```diff
@@ -99,7 +99,7 @@ export default ['i18n', function(i18n) {
                     disableChooseOption: true,
                     ngModel: 'user_type',
                     ngShow: 'current_user["is_superuser"]',
-                    ngDisabled: '!(user_obj.summary_fields.user_capabilities.edit || canAdd)'
+                    ngDisabled: 'user_obj.id == 1 || !(user_obj.summary_fields.user_capabilities.edit || canAdd)'
```

Comment:
We can't assume a default user. it is not necessarily the case that the user with an `id` of 1 is the last remaining admin user or even an admin user at all.

To detect whether or not there is only one admin user left, additional api calls will be needed. You'll need to make a request to the `/api/v2/users` endpoint with query params `{ is_superuser: True }`:

```
http GET /api/v2/users/?is_superuser=True`
```

If the count from that query is `1` _and_ the user you're attempting to modify is a super_user, you could show a warning.

The other thing to remember is that the _current user_ that is performing the update must be a super_user as well, otherwise they have no way of knowing how many total super users exist on the system in the first place. You will need to check for this condition before doing any of these other checks. The overall process is as follows:

```
if the user is modifying themselves:
  if the user is being changed from super user to non super-user:
    make a request to http GET /api/v2/users/?is_superuser=True`
    if the count from this request is 1:
      show a warning
else:
  do exactly the same thing as before
```

_Alternatively_ we could just always show the warning whenever a superuser tries to remove their own superuser privilege. That would be less complicated to implement:


```
if the user is modifying themselves:
  if the user is being changed from super user to non super-user:
      show a warning
else:
  do exactly the same thing as before
```